### PR TITLE
Add testing_version flag

### PR DESCRIPTION
**Pull request by lidizheng**
grpc/grpc#27373

Test run: https://fusion2.corp.google.com/invocations/cf986474-d6b4-4709-8a5d-a7144e841aa2

@dapengzhang0

### CODE REVIEW

**Comment by lidizheng** [comment]
Bump the PR.

**Review by dapengzhang0 — APPROVED** [review]

File: buildscripts/kokoro/xds_url_map.sh [hunk 1]
```diff
@@ -108,6 +108,7 @@ run_test() {
     --flagfile="${TEST_DRIVER_FLAGFILE}" \
     --kube_context="${KUBE_CONTEXT}" \
     --client_image="${CLIENT_IMAGE_NAME}:${GIT_COMMIT}" \
+    --testing_version=$(echo "$KOKORO_JOB_NAME" | sed -E 's|^grpc/java/([^/]+)/.*|\1|') \
```
Comment (ejona86) [inline buildscripts/kokoro/xds_url_map.sh:111]: Put this in double quotes, for safety?

**Comment by ejona86** [comment]
I'm not sure how to merge this, since it is a "backport" (but not a backport, more of just a copy), but now it is different than the original. I didn't realize that when I made my comment.

**Comment by lidizheng** [comment]
This is my bad. I was using the same ad-hoc script to generate commit and PR. This flag is used to allow test cases to skip for unsupported client/server. Today, Java supports all test cases. This PR is a preventive work for future gap, or deprecation.

**Pull request merged**